shared/actions/login/index.js

```diff
@@ -458,79 +368,99 @@ export function addANewDevice () : AsyncAction {
   }
 }
 
-function makeKex2IncomingMap (dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle) : incomingCallMapType {
+function makeKex2IncomingMap (dispatch, getState/*, provisionMethod, userPassTitle, userPassSubtitle*/) : incomingCallMapType {
   return {
-    'keybase.1.provisionUi.chooseProvisioningMethod': (param, response) => {
-      response.result(provisionMethod)
-    },
     'keybase.1.loginUi.getEmailOrUsername': (param, response) => {
-      const {username} = getState().login.userPass
-      if (!username) {
-        dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
-          const {username} = getState().login.userPass
-          response.result(username)
-        }))
-      } else {
-        response.result(username)
+      const props = {
+        onSubmit: usernameOrEmail => response.result(usernameOrEmail)
       }
+
+      dispatch(routeAppend({
+        parseRoute: {componentAtTop: {component: UsernameOrEmail, props}}
+      }))
+    },
+    'keybase.1.provisionUi.chooseDevice': ({devices}, response) => {
+      const props = {
+        devices,
+        onSelect: deviceID => {
+          const type: deviceRole = devices[devices.findIndex(d => d.deviceID === deviceID)].type
+          const role = {
+            mobile: Constants.codePageDeviceRoleExistingPhone,
+            computer: Constants.codePageDeviceRoleExistingComputer
```

Comment:
so the rpc takes in a string but has a comment that it needs to be libkb.DeviceTypeDesktop / deviceTypeMobile
https://github.com/keybase/client/blob/master/protocol/avdl/login.avdl#L29
We only pull in a subset of these constants so we don't actually have them currently.